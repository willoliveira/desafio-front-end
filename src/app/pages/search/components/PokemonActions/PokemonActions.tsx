import React, { FC } from 'react';
import { View, StyleSheet, Text, Button } from 'react-native';

export interface IPokemonActionsProps {
	pokemonData: any;
	onAddPokemon: () => void;
	onRemovePokemon: () => void;
}

export const PokemonActions: FC<IPokemonActionsProps> = ({ pokemonData, onAddPokemon, onRemovePokemon }: IPokemonActionsProps) => (
		<View style={styles.pokemonActionContainer}>
			{
				pokemonData
					? (
						<Button
							title="Remover"
							onPress={onRemovePokemon} />
					)
					: (
						<Button
							title="Adicionar"
							onPress={onAddPokemon} />
					)
			}
		</View>
)


const styles = StyleSheet.create({
	pokemonActionContainer: {
		position: 'absolute',
		top: 12,
		right: 0
	}
});
