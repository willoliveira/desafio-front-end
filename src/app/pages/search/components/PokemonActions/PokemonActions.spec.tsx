/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { PokemonActions, IPokemonActionsProps } from './PokemonActions';

describe('<PokemonActions />', () => {

	let actionsProps;
	beforeAll(() => {
		actionsProps = {
			onAddPokemon: () => jest.fn(),
			onRemovePokemon: () =>  jest.fn()
		}
	});

	describe('Rendering', () => {

		it('should match to snapshot - add button', () => {
			const component = shallow(<PokemonActions pokemonData={true} {...actionsProps} />);
			expect(component).toMatchSnapshot("PokemonActions snapshot");
		});

		it('should match to snapshot - remove button', () => {
			const component = shallow(<PokemonActions pokemonData={false} {...actionsProps} />);
			expect(component).toMatchSnapshot("PokemonActions snapshot");
		});
	});
});
