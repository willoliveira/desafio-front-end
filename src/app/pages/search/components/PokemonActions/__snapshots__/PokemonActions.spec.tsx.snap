// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonActions /> Rendering should match to snapshot - add button: PokemonActions snapshot 1`] = `
<View
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 12,
    }
  }
>
  <Button
    onPress={[Function]}
    title="Remover"
  />
</View>
`;

exports[`<PokemonActions /> Rendering should match to snapshot - remove button: PokemonActions snapshot 1`] = `
<View
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 12,
    }
  }
>
  <Button
    onPress={[Function]}
    title="Adicionar"
  />
</View>
`;
